tests/network/vmi_istio.go

```diff
@@ -0,0 +1,313 @@
+			)
+
+			curlCommand := func(serverIP string, port int) string {
+				return fmt.Sprintf("curl -sD - -o /dev/null http://%s:%d | head -n 1\n", serverIP, port)
+			}
+
+			generateExpectedHTTPReturnCodesRegex := func(shouldBeReachable bool) (string, error) {
+				var codes []string
+				if shouldBeReachable {
+					codes = append(codes, "200")
+				} else {
+					codes = append(codes, "5..")
+				}
+				return fmt.Sprintf("HTTP\\/[123456789\\.]{1,3}\\s(%s)", strings.Join(codes, "|")), nil
+			}
+
+			BeforeEach(func() {
+				serverVMI = libvmi.NewCirros(
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding([]v1.Port{}...)),
+				)
+
+				serverVMI, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(serverVMI)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(console.LoginToCirros(serverVMI)).To(Succeed())
+
+				By("Starting HTTP Server")
+				tests.StartHTTPServer(serverVMI, testPort)
+
+				By("Getting back the Server VMI IP")
+				serverVMI, err = virtClient.VirtualMachineInstance(serverVMI.Namespace).Get(serverVMI.Name, &metav1.GetOptions{})
+				ExpectWithOffset(1, err).ShouldNot(HaveOccurred())
+				serverVMIAddress = serverVMI.Status.Interfaces[0].IP
+			})
+
+			// Istio envoy may intercept the request, we need to check the header response, not just the
+			// return code of curl, because if envoy proxy forbids a request, it responds with 502 Bad Gateway code
+			It("Should be able to reach kubevirt.io", func() {
+				var expectedReturnCodes string
+				expectedReturnCodes, err = generateExpectedHTTPReturnCodesRegex(true)
+				Expect(err).ShouldNot(HaveOccurred())
+				err = console.SafeExpectBatch(vmi, []expect.Batcher{
+					&expect.BSnd{S: "\n"},
+					&expect.BExp{R: console.PromptExpression},
+					&expect.BSnd{S: curlCommand(serverVMIAddress, testPort)},
+					&expect.BExp{R: expectedReturnCodes},
+					&expect.BSnd{S: "echo $?\n"},
+					&expect.BExp{R: console.RetValue("0")},
+				}, 60)
+				Expect(err).ToNot(HaveOccurred())
+			})
+			Context("With Sidecar allowing only registered external services", func() {
+				var registryOnlySidecar *unstructured.Unstructured
+				BeforeEach(func() {
+					registryOnlySidecar = &unstructured.Unstructured{
+						Object: map[string]interface{}{
+							"apiVersion": "networking.istio.io/v1beta1",
+							"kind":       "Sidecar",
+							"metadata": map[string]interface{}{
+								"name": "registry-only-sidecar",
+							},
+							"spec": map[string]interface{}{
+								"outboundTrafficPolicy": map[string]interface{}{
+									"mode": "REGISTRY_ONLY",
+								},
+							},
+						},
+					}
+					_, err = virtClient.DynamicClient().Resource(sidecarRes).Namespace(tests.NamespaceTestDefault).Create(context.TODO(), registryOnlySidecar, metav1.CreateOptions{})
+					ExpectWithOffset(1, err).ShouldNot(HaveOccurred())
```

Comment:
with offset is not needed.

Reply:
https://github.com/kubevirt/kubevirt/pull/5200#discussion_r621903611